libbeat/reader/syslog/rfc3164_test.go

```diff
@@ -156,7 +156,7 @@ func TestParseRFC3164(t *testing.T) {
 				hostname: "test-host",
 				msg:      "this is the message",
 			},
-			wantErr: `validation error at position 5: parsing time "24-08-2003T05:14:15-07:00" as "2006-01-02T15:04:05.999999999Z07:00": cannot parse "8-2003T05:14:15-07:00" as "2006"`,
+			wantErr: `validation error at position 5: parsing time "24-08-2003T05:14:15-07:00" as "2006-01-02T15:04:05.999999999Z07:00": cannot parse "24-08-2003T05:14:15-07:00" as "2006"`,
```

Comment:
Why did this change? Are you sure this is a not a functional change in the parser that caused this? Was there a change in Go 1.20 that needed this?

I want to make sure this is not a hint that something has changed in how we parse values that we need to account for before we release this.